src/main/scala/mesosphere/marathon/api/v2/LeaderResource.scala

```diff
@@ -0,0 +1,27 @@
+package mesosphere.marathon.api.v2
+
+import javax.ws.rs.core.{ MediaType, Response }
+import javax.ws.rs.{ GET, DELETE, Path, Produces }
+
+import com.google.inject.Inject
+import mesosphere.chaos.http.HttpConf
+import mesosphere.marathon.api.RestResource
+import mesosphere.marathon.{ MarathonSchedulerService, MarathonConf }
+
+@Path("v2/leader")
+class LeaderResource @Inject() (
+    schedulerService: MarathonSchedulerService,
+    val config: MarathonConf with HttpConf)
+    extends RestResource {
+
+  @DELETE
+  @Produces(Array(MediaType.APPLICATION_JSON))
+  def delete(): Response = {
+    schedulerService.isLeader match {
+      case false => notFound(s"Marathon instance '${config.hostname}' is no leader")
```

Comment:
There is a contradiction also in the /v2/info resource: all requests are forwarded to the leader anyway. Hence, either the REST service responds, then it's the leader answering. Or the REST services doesn't respond, then there is no leader.